### ISSUE TITLE
Provide environment variables without `env.` prefix

### ISSUE THREAD

**Issue by jklukas** [tkrille/dropwizard-template-config#16]
If environment variables were injected directly into the data model (`${JAVA_HOME}`) rather than being accessed underneath the `env` variable (`${env.JAVA_HOME}`), the functionality defined in this package would become a superset of Dropwizard's built-in `EnvironmentVariableSubstitutor`. This would allow an easy "upgrade path" from Dropwizard's substitutor to this bundle as a project's needs get more complex.

@tkrille - What do you think of adding this? Could we potentially make this a default behavior? Or would you want to make users turn it on explicitly through a new option, say a boolean `flattenEnvironment`?

**Comment by jklukas** [comment]
> the functionality defined in this package would become a superset of Dropwizard's built-in EnvironmentVariableSubstitutor

I just realized the above statement was overly strong, as the EnvironmentVariableSubstitutor has a syntax for defaults which differs from freemarker (`:-` vs. `!`), but transitioning between the two would still be easier if it were possible to skip the `env.` prefix.

**Comment by tkrille** [comment]
I really like the idea. I always felt, that the syntax for accessing variables is overly verbose, too. The issue with this is, that this bundle not only support environment variables, but system properties, too. One possible solution would be to flatten and merge both system properties and environment variables with system properties having a higher or lower precedence. I think the details just have to be figured out, but the idea is very good und we should pursue it.

**Comment by tkrille** [comment]
What do you think? What should have higher precedence? I think we can make this change backwards compatible, which should work as long as no one defines variables named `env`, or `sys`.

**Comment by tkrille** [comment]
If we come up with a working solutions quick, this might be included in 1.3.0 too.

**Comment by jklukas** [comment]
> What do you think? What should have higher precedence?

I've never had a need to inject a system property into a config, so I'd certainly be biased to give environment variables higher precedence.

I believe that implementing that precedence would be quite straightforward, loading items into the data model in increasing order of precendence, letting later additions overwrite any earlier definitions:
```
dataModel.putAll(systemPropertiesProvider.getSystemProperties());
dataModel.putAll(environmentProvider.getEnvironment());
dataModel.put("sys", systemPropertiesProvider.getSystemProperties());
dataModel.put("env", environmentProvider.getEnvironment());
```

**Comment by tkrille** [comment]
This looks really good. Do you want to implement it? :)

**Comment by tkrille** [comment]
I agree with you about system properties.

**Comment by tkrille** [comment]
Closed with #17
